### PR TITLE
Petsc (sundials,hdf5,hwloc,suite-sparse,netcdf,fftw)

### PR DESCRIPTION
**Pull request by davydden**
Added sundials and hdf5 dependencies.

### CODE REVIEW

**Comment by davydden** [comment]
before merging, i though one could output test results via `ohai`. Perhaps taking output stored in logs of `make test` (excerpt):
```
C/C++ example src/snes/examples/tutorials/ex19 run successfully with 1 MPI process
C/C++ example src/snes/examples/tutorials/ex19 run successfully with 2 MPI processes
Fortran example src/snes/examples/tutorials/ex5f run successfully with 1 MPI process
```
?

is there a command to get the current log folder? Or any alternative ways to achieve it?

**Comment by dpo** [comment]
Have a look at [`octave.rb`](https://github.com/Homebrew/homebrew-science/blob/master/octave.rb#L183); the output of `make check` is `tee`'d to a log file that is later `grep`ped for indications of failure. I think a summary of failures, if there are any, could be good.

File: petsc.rb [hunk 1]
```diff
@@ -31,8 +31,16 @@ class Petsc < Formula
   depends_on "metis"        => :recommended
   depends_on "parmetis"     => :recommended
   depends_on "scalapack"    => :recommended
-  depends_on "mumps"        => :recommended
+  depends_on "mumps"        => :recommended #mumps is built with mpi by default
```
Comment (dpo) [inline petsc.rb]: Just a couple minor comments. Please use a space after `#`.

File: petsc.rb [hunk 2]
```diff
@@ -83,9 +95,16 @@ def install
     args_real = ["--prefix=#{prefix}/#{arch_real}",
                  "--with-scalar-type=real"]
     args_real << "--with-hypre-dir=#{oprefix("hypre")}" if build.with? "hypre"
+    args_real << "--with-sundials-dir=#{oprefix("sundials")}" if build.with? "sundials"
+    args_real << "--with-hwloc-dir=#{oprefix("hwloc")}" if build.with? "hwloc"
     system "./configure", *(args + args_real)
     system "make", "all"
-    system "make", "test" if build.with? "check"
+    if build.with? "check"
+      log_name = "make-check-"+arch_real+".log"
```
Comment (dpo) [inline petsc.rb]: Please use spaces around `+`.

File: petsc.rb [hunk 3]
```diff
@@ -94,7 +113,12 @@ def install
                   "--with-scalar-type=complex"]
     system "./configure", *(args + args_cmplx)
     system "make", "all"
-    system "make", "test" if build.with? "check"
+    if build.with? "check"
+      log_name = "make-check-"+arch_complex+".log"
```
Comment (dpo) [inline petsc.rb]: Same here.

**Comment by dpo** [comment]
@BrewTestBot test this please

File: petsc.rb [hunk 4]
```diff
@@ -83,9 +95,16 @@ def install
     args_real = ["--prefix=#{prefix}/#{arch_real}",
                  "--with-scalar-type=real"]
     args_real << "--with-hypre-dir=#{oprefix("hypre")}" if build.with? "hypre"
+    args_real << "--with-sundials-dir=#{oprefix("sundials")}" if build.with? "sundials"
+    args_real << "--with-hwloc-dir=#{oprefix("hwloc")}" if build.with? "hwloc"
     system "./configure", *(args + args_real)
     system "make", "all"
-    system "make", "test" if build.with? "check"
+    if build.with? "check"
+      log_name = "make-check-" + arch_real+".log"
```
Comment (dpo) [inline petsc.rb]: Some spaces still missing here.

File: petsc.rb [hunk 5]
```diff
@@ -94,7 +113,12 @@ def install
                   "--with-scalar-type=complex"]
     system "./configure", *(args + args_cmplx)
     system "make", "all"
-    system "make", "test" if build.with? "check"
+    if build.with? "check"
+      log_name = "make-check-" + arch_complex+".log"
```
Comment (dpo) [inline petsc.rb]: and here.

**Pull request closed without merging**